src/accelerate/accelerator.py

```diff
@@ -2300,6 +2300,12 @@ def clip_grad_norm_(self, parameters, max_norm, norm_type=2):
                     xm.all_reduce("sum", gradients, scale=1.0 / self.num_processes)
                     # Set is_xla_gradients_synced to True to avoid all-reduce twice in the AcceleratedOptimizer step.
                     acc_opt.gradient_state.is_xla_gradients_synced = True
+            if os.environ.get("ACCELERATE_USE_FSDP", "false") == "true":
```

Comment:
Thanks. There's a dark pattern I worry about here with the XLA integration because we really *should not* be relying on environmental variables for things. We should be relying only on the `AcceleratorState` or `PartialState`.

E.g. the expectation is this should be:

- `self.state.distributed_type == DistributedType.FSDP`


This PR is fine, however this means that we need to introduce probably a separate `DeviceType` as part of the `PartialState`, which I was already considering. cc @SunMarc (not a concern wrt this PR, just noting down for me to remember later etc)